### PR TITLE
Serf: add Python dependency, sort dependencies

### PR DESCRIPTION
**Pull request by christoph-conrads**

### CODE REVIEW

File: var/spack/repos/builtin/packages/serf/package.py [hunk 1]
```diff
@@ -19,13 +19,13 @@ class Serf(SConsPackage):
     variant('debug', default=False,
             description='Enable debugging info and strict compile warnings')
 
-    depends_on('scons@2.3.0:', type='build')
-
     depends_on('apr')
     depends_on('apr-util')
     depends_on('openssl')
-    depends_on('zlib')
+    depends_on('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/serf/package.py]: Should this be type build? Does it link to the Python libs/headers? Is it used at run-time?

Comment (christoph-conrads) [reply var/spack/repos/builtin/packages/serf/package.py]: I do not know. The README does not even mention Python (or the available CMake build), there is no file called `INSTALL` or anything similar, and I am not familiar with SCons.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/serf/package.py]: The build log doesn't mention Python in any of the link lines and the resulting shared object library doesn't seem to link to Python, so I think we can make it `type='build'`.

Comment (christoph-conrads) [reply var/spack/repos/builtin/packages/serf/package.py]: This sounds reasonable. I updated the build.

File: var/spack/repos/builtin/packages/serf/package.py [hunk 2]
```diff
@@ -19,13 +19,13 @@ class Serf(SConsPackage):
     variant('debug', default=False,
             description='Enable debugging info and strict compile warnings')
 
-    depends_on('scons@2.3.0:', type='build')
-
     depends_on('apr')
     depends_on('apr-util')
     depends_on('openssl')
-    depends_on('zlib')
+    depends_on('python', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/serf/package.py]:
```suggestion
    depends_on('python+pythoncmd', type='build')
```
Might want to add this until #25408 gets resolved. Or you could add a `patch()` method to fix it in this PR.

Comment (christoph-conrads) [reply var/spack/repos/builtin/packages/serf/package.py]: Thanks for your proposal; I did not know this variant before. It was added.

**Review by adamjstewart — APPROVED** [review]

**Comment by christoph-conrads** [comment]
Closing the issue because the build update fixed it for me.

**Pull request merged**